src/ConditionalApplyPlugin.groovy

```diff
@@ -1,18 +1,22 @@
 import static TerraformEnvironmentStage.CONFIRM
 import static TerraformEnvironmentStage.APPLY
+import static TerraformDestroyStage.DESTROY
 
 public class ConditionalApplyPlugin implements TerraformEnvironmentStagePlugin {
 
     private String branch
 
     ConditionalApplyPlugin() {
-        branch = 'master'
+        branch = 'destroy_test' // TODO: CHANGE THIS BACK TO MASTER!
     }
 
     @Override
     public void apply(TerraformEnvironmentStage stage) {
         stage.decorateAround(CONFIRM, onlyOnExpectedBranch())
         stage.decorateAround(APPLY, onlyOnExpectedBranch())
+        if (stage instanceof TerraformDestroyStage){
+            stage.decorateAround(DESTROY, onlyOnExpectedBranch())
+        }
     }
```

Comment:
There's two ways you can avoid the `instanceof TerraformDestroyStage`.
1. If you define a `public void apply(TerraformDestroyStage stage)` method, Groovy will be smart enough to use that one for TerraformDestroyStages, and use this one for normal TerraformEnvironmentStages.  That is, you let the compiler implicitly do the `if instanceof TerraformDestroy` on your behalf.
2. We don't care whether or not you're a normal TerraformEnvironmentStage or a TerraformDestroyStage.  Either way, we do the same thing, and decorate the CONFIRM and APPLY.  This means that in a TerraformDestroyStage 'APPLY' is now the time that we run the `terraform destroy` command instead of the `terraform apply` command.